tests/manage/mcg/test_namespace.py

```diff
@@ -30,83 +31,185 @@ class TestNamespace(MCGTest):
     @tier1
-    def test_namespace_resource_creation(self, ns_resource_factory):
+    def test_namespace_store_creation(self, ns_store_factory):
         """
-        Test namespace resource creation using the MCG RPC.
+        Test namespace store creation using the MCG CRDs.
         """
-        # Create the namespace resource and verify health
-        ns_resource_factory()
+        # Create the namespace store and verify health
+        ns_store_factory()
 
+    @skipif_ocs_version("<4.7")
     @tier1
     @pytest.mark.parametrize(
-        argnames=["platform"],
+        argnames=["platform", "interface", "namespace_policy_type"],
         argvalues=[
             pytest.param(
-                constants.AWS_PLATFORM, marks=pytest.mark.polarion_id("OCS-2256")
+                constants.AWS_PLATFORM,
+                "OC",
+                constants.NAMESPACE_POLICY_TYPE_SINGLE,
+                marks=pytest.mark.polarion_id("OCS-2256"),
+            ),
+            pytest.param(
+                constants.AZURE_PLATFORM,
+                "OC",
+                constants.NAMESPACE_POLICY_TYPE_SINGLE,
+                marks=pytest.mark.polarion_id("OCS-2409"),
+            ),
+            pytest.param(
+                constants.AWS_PLATFORM,
+                "OC",
+                constants.NAMESPACE_POLICY_TYPE_MULTI,
             ),
             pytest.param(
-                constants.AZURE_PLATFORM, marks=pytest.mark.polarion_id("OCS-2409")
+                constants.AZURE_PLATFORM,
+                "OC",
+                constants.NAMESPACE_POLICY_TYPE_MULTI,
             ),
         ],
     )
     def test_namespace_bucket_creation(
-        self, ns_resource_factory, bucket_factory, platform
+        self,
+        ns_store_factory,
+        bucket_factory,
+        bucket_class_factory,
+        platform,
+        interface,
+        namespace_policy_type,
     ):
         """
-        Test namespace bucket creation using the MCG RPC.
+        Test namespace bucket creation using the MCG CRDs.
         """
         # Create the namespace resource and verify health
-        ns_resource_name = ns_resource_factory(platform=platform)[1]
+        ns_store_name = ns_store_factory(platform=platform)[1]
 
         # Create the namespace bucket on top of the namespace resource
+        bucketclass_dict = {
+            "interface": interface,
+            "namespace_policy_dict": {"type": namespace_policy_type},
+        }
+        if namespace_policy_type == constants.NAMESPACE_POLICY_TYPE_SINGLE:
+            bucketclass_dict["namespace_policy_dict"]["resource"] = ns_store_name
+        elif namespace_policy_type == constants.NAMESPACE_POLICY_TYPE_MULTI:
+            bucketclass_dict["namespace_policy_dict"]["read_resources"] = [
+                ns_store_name
+            ]
+            bucketclass_dict["namespace_policy_dict"]["write_resource"] = ns_store_name
+
+        bucketclass = bucket_class_factory(bucketclass_dict)
```

Comment:
As part of #3437, all the bucket setup (including bucketclasses, backingstores and ULS) is happening behind the scenes as part of `bucket_factory`.
Please migrate the NSS creation/deletion/manipulation logic to work with this flow